### ISSUE TITLE
Problem with file ownership in data only container when using volume

### ISSUE THREAD

**Issue by stefanleh** [moby/moby#30098]
**When using volume of data only container the ownership is not respected**

**Steps to reproduce the issue:**
1. docker create -u 1000:1000 -v "/home/jenkins/.gradle/caches/modules-2/files-2.1" --name jenkins-gradle-data stefanlehmann/centos-jenkins-docker-slave:withdockerclient
2. docker run -it --volumes-from=jenkins-gradle-data:rw stefanlehmann/centos-jenkins-docker-slave:withdockerclient bash
3. grep 1000 /etc/passwd
```jenkins:x:1000:1000::/home/jenkins:/bin/bash```
4.  ls -al /home/jenkins/

**Describe the results you received:**

```
[root@91d10cf791c6 jenkins]# ls -al
total 12
drwx------ 3 jenkins jenkins  77 Jan 12 15:03 .
drwxr-xr-x 3 root    root     21 Dec 15 23:56 ..
-rw-r--r-- 1 jenkins jenkins  18 Dec  6 23:19 .bash_logout
-rw-r--r-- 1 jenkins jenkins 193 Dec  6 23:19 .bash_profile
-rw-r--r-- 1 jenkins jenkins 231 Dec  6 23:19 .bashrc
drwxr-xr-x 3 root    root     20 Jan 12 15:03 .gradle
```

**Describe the results you expected:**

```
[root@91d10cf791c6 jenkins]# ls -al
total 12
drwx------ 3 jenkins jenkins  77 Jan 12 15:03 .
drwxr-xr-x 3 root    root     21 Dec 15 23:56 ..
-rw-r--r-- 1 jenkins jenkins  18 Dec  6 23:19 .bash_logout
-rw-r--r-- 1 jenkins jenkins 193 Dec  6 23:19 .bash_profile
-rw-r--r-- 1 jenkins jenkins 231 Dec  6 23:19 .bashrc
drwxr-xr-x 3 jenkins jenkins 20 Jan 12 15:03 .gradle
```

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

```
[slehmann@docker-host-fkb ~]$ docker version
Client:
 Version:      1.12.5
 API version:  1.24
 Go version:   go1.6.4
 Git commit:   7392c3b
 Built:        Fri Dec 16 02:23:59 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.5
 API version:  1.24
 Go version:   go1.6.4
 Git commit:   7392c3b
 Built:        Fri Dec 16 02:23:59 2016
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
[slehmann@docker-host-fkb ~]$ docker info
Containers: 12
 Running: 8
 Paused: 0
 Stopped: 4
Images: 19
Server Version: 1.12.5
Storage Driver: devicemapper
 Pool Name: docker-8:17-6433140-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: xfs
 Data file: /dev/centos/docker-data
 Metadata file: /dev/centos/docker-metadata
 Data Space Used: 31.57 GB
 Data Space Total: 139.6 GB
 Data Space Available: 108 GB
 Metadata Space Used: 29.92 MB
 Metadata Space Total: 4.295 GB
 Metadata Space Available: 4.265 GB
 Thin Pool Minimum Free Space: 13.96 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Library Version: 1.02.135-RHEL7 (2016-09-28)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: null overlay host bridge
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Security Options: seccomp
Kernel Version: 3.10.0-514.2.2.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
CPUs: 8
Total Memory: 31.16 GiB
Name: docker-host-fkb
ID: CF5W:NXBN:LMPW:5GLE:A4O3:NYN4:QJQN:LTXH:K5JA:EE3V:OI2X:FW5D
Docker Root Dir: /docker2/docker_instance1/graph
Debug Mode (client): false
Debug Mode (server): false
Username: stefanlehmann
Registry: https://index.docker.io/v1/
Insecure Registries:
 unic-dev-dockerreg.int.somedomain.com
 127.0.0.0/8

```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Comment by thaJeztah** [comment]
This is the expected behavior; when defining a volume for a specific path in the container, docker will look for the specified path inside the container (in this case `/home/jenkins/.gradle/caches/modules-2/files-2.1`). If the path exists inside the container, it's used as a mount-point and the volume will be mounted at that location. If the path does not exist, the path is created (as `root`, as it doesn't know what permissions should be used) and used as mount point after it's created.

While it _could_ be possible to `chown` the path to the current container user after creating, I don't think it _should_ (it would then be _assuming_ that you always want it to be owned by the current user, but there's many situations where that would not be desirable).

The best option here is to make sure the directory exists inside the container image with the right ownership/permissions, before starting the container
